Added task 7

luminorena opened the pull request.

luminorena committed:
Added task 7

DanilShatov commented on basic-7/src/main/java/ru/otus/Car.java:
```diff
@@ -0,0 +1,44 @@
+package ru.otus;
+
+public class Car extends Human implements Transport {
```
почему машина является наследником человека? Логически не понятно

luminorena committed:
Fixed task 7

luminorena committed:
Fixed task 7 #2

FlameXander commented on basic-7/src/main/java/ru/otus/Human.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public boolean move(int distance, TerritoryType territoryType) {
+        var remainingPower = humanPower - Math.random() * distance;
```
Почему человек устает сидя в машине?

FlameXander commented on basic-7/src/main/java/ru/otus/Human.java:
```diff
@@ -0,0 +1,50 @@
+       }
+        if (currentTransport != null) {
+            var res = currentTransport.move(distance, territoryType);
+            if (res) {
```
Зачем человек выходит из "машины"?

FlameXander commented on basic-7/src/main/java/ru/otus/Human.java:
```diff
@@ -0,0 +1,50 @@
+            System.out.println("У человека нет сил пройти расстояние " + distance);
+            return false;
+        } else {
+            System.out.println("Человек " + name + " прошел расстояние " + distance);
```
А почему его сила именно тут не уменьшилась?

luminorena committed:
Fixed task 7 #3
renamed variable humanPower to personPower
added message with remaining person power

luminorena committed:
Fixed task 7 #4
changed position for tired human code
